src/typing.py

```diff
@@ -991,14 +968,26 @@ def __new__(cls, name, bases, namespace,
         initial_bases = bases
         if extra is not None and type(extra) is abc.ABCMeta and extra not in bases:
             bases = (extra,) + bases
-        bases = tuple(_gorg(b) if isinstance(b, GenericMeta) else b for b in bases)
+        new_bases = []
+        for base in bases:
+            if isinstance(base, GenericMeta):
+                bextra = getattr(base, '__extra__', None)
+                if (extra and bextra and not origin and bextra not in bases and
+                        type(bextra) is abc.ABCMeta):
```

Comment:
The point is this is not totally equivalent. This makes more "aggressive" erasure for classes with ``__extra__``, this shortens MROs for special typing types. Compare:
Before:
```python
>>> pprint(List[int].__mro__)
(typing.List[int],
 typing.List,
 <class 'list'>,
 typing.MutableSequence,
 <class 'collections.abc.MutableSequence'>,
 typing.Sequence,
 <class 'collections.abc.Sequence'>,
 typing.Reversible,
 <class 'collections.abc.Reversible'>,
 typing.Collection,
 <class 'collections.abc.Collection'>,
 <class 'collections.abc.Sized'>,
 typing.Iterable,
 <class 'collections.abc.Iterable'>,
 typing.Container,
 <class 'collections.abc.Container'>,
 typing.Generic,
 <class 'object'>)
```
and after:
```python
(typing.List[int],
 typing.List,
 <class 'list'>,
 <class 'collections.abc.MutableSequence'>,
 <class 'collections.abc.Sequence'>,
 <class 'collections.abc.Reversible'>,
 <class 'collections.abc.Collection'>,
 <class 'collections.abc.Sized'>,
 <class 'collections.abc.Iterable'>,
 <class 'collections.abc.Container'>,
 <class 'object'>)
```